File: TFG/hotftp_api/src/application/use-cases/monitoring/get-health-summary.ts
import type { HealthSummary } from '../../../domain/entities/health-summary.js';
import type { MonitoringRepository } from '../../../domain/repositories/monitoring-repository.js';
import type { ProfileRepository } from '../../../domain/repositories/profile-repository.js';

export class GetHealthSummary {
  constructor(
    private readonly profileRepository: ProfileRepository,
    private readonly monitoringRepository: MonitoringRepository,
  ) {}

  async execute(ownerId: string): Promise<HealthSummary> {
    const [summary, profiles] = await Promise.all([
      this.monitoringRepository.getHealthSummary(ownerId),
      this.profileRepository.list(ownerId),
    ]);

    return {
      ...summary,
      totalProfiles: profiles.length,
    };
  }
}

File: TFG/ftp_tfg/frontend/lib/domain/usecases/get_health_summary.dart
import '../entities/system_health_summary.dart';
import '../repositories/monitoring_repository.dart';
import '../interfaces/i_get_health_summary_use_case.dart';

class GetHealthSummary implements IGetHealthSummaryUseCase {
  final MonitoringRepository repository;

  GetHealthSummary(this.repository);

  @override
  Future<SystemHealthSummary> execute(String ownerId) =>
      repository.getHealthSummary(ownerId);
}




